pkg/proxy/endpoints.go

```diff
@@ -407,20 +407,19 @@ func detectStaleConntrackEntries(oldEndpointsMap, newEndpointsMap EndpointsMap,
 		}
 
 		for _, ep := range epList {
-			// If the old endpoint wasn't Ready then there can't be stale
+			// If the old endpoint wasn't Serving then there can't be stale
 			// conntrack entries since there was no traffic sent to it.
-			if !ep.IsReady() {
+			if !ep.IsServing() {
 				continue
 			}
 
 			deleted := true
 			// Check if the endpoint has changed, including if it went from
-			// ready to not ready. If it did change stale entries for the old
+			// serving to not serving. If it did change stale entries for the old
 			// endpoint have to be cleared.
 			for i := range newEndpointsMap[svcPortName] {
 				if newEndpointsMap[svcPortName][i].String() == ep.String() &&
-					newEndpointsMap[svcPortName][i].IsReady() == ep.IsReady() &&
-					newEndpointsMap[svcPortName][i].GetIsLocal() == ep.GetIsLocal() {
+					newEndpointsMap[svcPortName][i].IsServing() == ep.IsServing() {
```

Comment:
@danwinship I don't understand this change, so previously we were detecting "old the endpoints were ready and new endpoints are ready and are local", and now we move to "old endpoints were serving and new endpoints are still serving".

Why do we remove the local condition?

If one endpoint is Serving and Terminating only will receive traffic if there are not more ready endpoints, do we want to perform additional conntracl flushes?

Reply:
I guess the **String()** comparison takes care of the local part. If both the endpoints have the same string representation (IP:PORT) then they are the same only,  both of them will be either local or remote and **GetIsLocal()** comparison will always be true, redundant here.

Reply:
oh, I somehow assumed this was only considering local endpoints, makes sense.

what about 

> If one endpoint is Serving and Terminating only will receive traffic if there are not more ready endpoints, do we want to perform additional conntracl flushes?

Reply:
> If one endpoint is Serving and Terminating only will receive traffic if there are not more ready endpoints, do we want to perform additional conntracl flushes?

The NAT conntrack entry will be flushed independently for every endpoint when it transitions from serving to not serving. And if the service is left with zero serving endpoints, as soon as any new endpoint becomes active **NewlyActiveUDPServices** flow will clear all the conntrack entries for that service. 

I guess we don't need any additional conntrack flushes.